near/omni-bridge/src/lib.rs

```diff
@@ -555,11 +599,149 @@ impl Contract {
             )
             .into()
         } else {
-            self.process_fin_transfer_to_other_cahin(predecessor_account_id, transfer_message);
+            self.process_fin_transfer_to_other_chain(predecessor_account_id, transfer_message);
             PromiseOrValue::Value(destination_nonce)
         }
     }
 
+    fn fast_fin_transfer(
+        &mut self,
+        token_id: AccountId,
+        amount: U128,
+        fast_fin_transfer_msg: FastFinTransferMsg,
+    ) -> PromiseOrValue<U128> {
+        let fast_transfer = FastTransfer {
+            token_id: token_id.clone(),
+            recipient: fast_fin_transfer_msg.recipient.clone(),
+            amount: U128(amount.0 + fast_fin_transfer_msg.fee.fee.0),
+            fee: fast_fin_transfer_msg.fee,
+            transfer_id: fast_fin_transfer_msg.transfer_id,
+            msg: fast_fin_transfer_msg.msg,
+        };
+
+        if let OmniAddress::Near(recipient) = fast_fin_transfer_msg.recipient {
+            let storage_deposit_amount = fast_fin_transfer_msg
+                .storage_deposit_amount
+                .unwrap_or_default();
+            if storage_deposit_amount > 0 {
+                self.update_storage_balance(
+                    fast_fin_transfer_msg.relayer.clone(),
```

Comment:
The relayer can be any arbitrary account; we do not check whether it matches the sender_id that transfers the tokens. As a result, we can force any arbitrary account to pay us for storage.

For example, we can make a transfer of a small amount (eps) of funds, where eps << storage_deposit, find an account with sufficient storage, and force it to top up our deposit for the token.

Reply:
Fixed with signer_id usage